src/main/java/org/quiltmc/enigma_plugin/index/simple_type_single/SimpleTypeFieldNamesRegistry.java

```diff
@@ -182,6 +232,145 @@ public Entry(String type, String localName, String staticName) {
 		}
 	}
 
-	public record Name(String local, String staticName) {
+	public record Name(String local, String constant) {
+	}
+
+	public sealed interface Inherit {
+		Inherit DEFAULT = None.INSTANCE;
+
+		String KEY = "inherit";
+		String TYPE_KEY = "type";
+
+		String MISSING_REQUIREMENT_MESSAGE_TEMPLATE = "%s requires a %s";
+
+		static Either<? extends Inherit, String> read(JsonReader reader) throws IOException {
+			switch (reader.peek()) {
+				case BOOLEAN -> {
```

Comment:
i'd prefer the boolean logic was removed in favor of directly naming the scheme, i think that inherit: true has always been confusing and i'd like to entirely remove it